File: goal-keeper/src/data/dto/PlayerDto.ts
import { JsonObject, JsonProperty } from 'json2typescript';
import DateConverter from "@/serializer/converter/DateConverter";

@JsonObject('PlayerDto')
export default class PlayerDto {

    @JsonProperty('id', String)
    public id: string | undefined = undefined;

    @JsonProperty('givenName', String)
    public givenName: string | undefined = undefined;

    @JsonProperty('surname', String)
    public surname: string | undefined = undefined;

    @JsonProperty('position', String)
    public position: string | undefined = undefined;

    @JsonProperty('shirtNumber', Number)
    public shirtNumber: number | undefined = undefined;

    @JsonProperty('preferredFoot', String)
    public preferredFoot: string | undefined = undefined;

    @JsonProperty('dateOfBirth', DateConverter, true)
    public dateOfBirth: Date | undefined = undefined;

    static create(player: PlayerDto) {
        return {
            id: player.id,
            givenName: player.givenName,
            surname: player.surname,
            position: player.position,
            shirtNumber: player.shirtNumber,
            preferredFoot: player.preferredFoot,
            dateOfBirth: player.dateOfBirth
        }
    }
}

File: goal-keeper-backend/src/main/java/com/goalkeeper/api/dto/PlayerDto.java
package com.goalkeeper.api.dto;

import com.fasterxml.jackson.annotation.JsonRootName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.UUID;

@Builder
@Data
@JsonRootName("player")
@NoArgsConstructor
@AllArgsConstructor
public class PlayerDto {

    private UUID id;

    @NotBlank
    @Size(min = 3, max = 20)
    private String givenName;

    @NotBlank
    @Size(max = 50)
    private String surname;

    @NotNull
    private Integer shirtNumber;

    private String position;
    private String preferredFoot;
    private LocalDateTime dateOfBirth;
    private Boolean active;


}
